client.go

```diff
@@ -1159,13 +1159,16 @@ func normaliseError(err error) error {
 }
 
 func unmarshalStatus(id uint32, data []byte) error {
+	var lang string
 	sid, data := unmarshalUint32(data)
 	if sid != id {
 		return &unexpectedIDErr{id, sid}
 	}
 	code, data := unmarshalUint32(data)
-	msg, data := unmarshalString(data)
-	lang, _, _ := unmarshalStringSafe(data)
+	msg, data, err := unmarshalStringSafe(data)
```

Comment:
Please change this to
```
        msg, data, err := unmarshalString(data)
        if err != nil {
                return err
        }
        lang, _, _ := unmarshalStringSafe(data)
```
My reading of https://tools.ietf.org/html/draft-ietf-secsh-filexfer-02 indicates that the message is not optional.